src/mysql_k8s_helpers.py

```diff
@@ -256,108 +253,75 @@ def create_custom_config_file(self, report_host: str) -> None:
             raise MySQLCreateCustomConfigFileError()
 
     @retry(
-        retry=retry_if_result(lambda x: not x),
+        retry=retry_if_exception_type(MySQLWaitUntilUnitRemovedFromClusterError),
         stop=stop_after_attempt(10),
         wait=wait_fixed(60),
     )
-    def _wait_till_all_members_are_online(self) -> None:
-        """Wait until all members of the cluster are online.
+    def _wait_until_unit_removed_from_cluster(self, unit_address: str) -> None:
```

Comment:
Could not find this being used anywhere. I'm missing something?

Reply:
Great catch! It was designed to be called in the function that forcefully removed a unit from the cluster. Added the function call in 0a939bd6332579e0fa0af593567f1b67265b4e8b